globals.go

```diff
@@ -120,6 +121,9 @@ var (
 	// be thread safe and non-blocking.
 	ErrorHandler func(err error)
 
+	// ExitFunc is the fucntion called to exit the application, defaults to `os.Exit()`
+	ExitFunc exitFunc = os.Exit
```

Comment:
I see a real problem here, with the implementation

You are doing this change for test purpose, but you are introducing a global exiter that anyone can overload.

It shouldn't be global and exported.

This would lead to unpredictable behavior with tests, but also with code.

Think about someone anywhere in log that would do a logger.ExitFunc = func(_ int) {}